openpype/hosts/max/plugins/publish/published_files_tycache_object.py

```diff
@@ -0,0 +1,37 @@
+import pyblish.api
+from pymxs import runtime as rt
+
+
+class PublishedFilesTycacheObject(pyblish.api.InstancePlugin):
+    """Load Published Files for Created TyCache Object."""
+
+    order = pyblish.api.IntegratorOrder + 0.89
+    label = "Published files for tycache Object"
+    hosts = ["max"]
+    families = ["tycache"]
+
+    def process(self, instance):
+        tycache_attrs = instance.data["tyc_attrs"]
+        tycache_Attributes = tycache_attrs["tycacheAttributes"]
+        if "tycacheCreateObjectIfNotCreated" not in tycache_Attributes:
+            self.log.debug("Attribute of 'Create Tycache Object' and "
+                           "'Only if one does not already exists' "
+                           "not being enabled. \n\n"
+                           "Skipping using published file as tyCache Object.")
+            return
+
+        published_path = None
+        tycache_object_name = tycache_attrs["tycacheObjectName"]
+        representation = instance.data["representations"]
+        for repre in representation:
+            published_path = repre["published_path"]
+
+        if "$(tyFlowName)" in tycache_object_name:
+            name = next(member.name for member in instance.data["members"])
+            tycache_object_name = tycache_object_name.replace(
+                "$(tyFlowName)", name)
+            tycache_object_node = rt.GetNodeByName(tycache_object_name)
+            if tycache_object_node:
+                tycache_object_node.filename = (
+                    published_path.replace("\\", "/")
+                )
```

Comment:
The loaded tyflow object is now _not_ managed by the pipeline as a "Load" from the loader would. Is that intended behavior?

Reply:
TyCache supports the behavior that it will create the object right after the publish. And it would use the staging directory but as we know it is a temp path so the tycache object would error out if the path does not exist anymore. So this PR is to ensure the tycache object can load the published files.

Reply:
> TyCache supports the behavior that it will create the object right after the publish.

I understand that TyCache supports its - but why are we supporting that in OpenPype? Is there any good reason for an artist on publish to do that?